src/Renderer/ThreeExtended/GlobeControls.js

```diff
@@ -770,210 +460,84 @@ function GlobeControls(view, target, radius, options = {}) {
-                    type: CONTROL_EVENTS.PAN_CHANGED,
-                });
-            }
-
-            const previousRange = snapShotSpherical.radius;
-            const newRange = this.getRange();
-            if (Math.abs(newRange - previousRange) / previousRange > 0.001) {
-                this.dispatchEvent({
-                    type: CONTROL_EVENTS.RANGE_CHANGED,
-                    previous: { range: previousRange },
-                    new: { range: newRange },
-                });
-            }
-
-            if (cameraTargetOnGlobe.position.distanceTo(previousCameraTargetOnGlobe) / spherical.radius > delta) {
-                this.dispatchEvent({
-                    type: CONTROL_EVENTS.CAMERA_TARGET_CHANGED,
-                    previous: { cameraTarget: new Coordinates(this._view.referenceCrs, previousCameraTargetOnGlobe) },
-                    new: { cameraTarget: new Coordinates(this._view.referenceCrs, cameraTargetOnGlobe.position) },
-                });
-            }
-            snapShotSpherical.copy(spherical);
-        }
-
-        state = this.states.NONE;
-        lastRotation = [];
-        if (enableTargetHelper) {
-            this._view.notifyChange(this.camera);
-        }
-    };
+    this.update = update;
 
     // Update helper
-    var updateHelper = enableTargetHelper ? function updateHelper(position, helper) {
+    const updateHelper = enableTargetHelper ? function updateHelper(position, helper) {
         positionObject(position, helper);
         this._view.notifyChange(this.camera);
     } : function empty() {};
 
-    this.getPickingPositionOnSphere = function getPickingPositionOnSphere() {
-        return tSphere.picking.position;
-    };
-
-    // Update radius's sphere : the sphere must cross the point
-    // Return intersection with mouse and sphere
-    var updateSpherePicking = (function getUpdateSpherePicking() {
-        var mouse = new THREE.Vector2();
-        var ray = new THREE.Ray();
+    const raycaster = new THREE.Raycaster();
+    function onMouseMove(event) {
+        if (player.isPlaying()) {
+            player.stop();
+        }
+        if (this.enabled === false) return;
 
-        return function updateSpherePicking(point, screenCoord) {
-            tSphere.setRadius(point.length());
+        cameraTarget.needUpdate = true;
 
-            mouse.x = (screenCoord.x / sizeRendering.width) * 2 - 1;
-            mouse.y = -(screenCoord.y / sizeRendering.height) * 2 + 1;
+        event.preventDefault();
+        const coords = view.eventToViewCoords(event);
 
-            snapShotCamera.updateRay(ray, mouse);
-            // pick position on tSphere
-            const its = tSphere.intersectWithRay(ray);
-            if (its != undefined) {
-                tSphere.picking.position.copy(its);
-                tSphere.picking.normal = tSphere.picking.position.clone().normalize();
+        if (state === this.states.ORBIT || state === this.states.PANORAMIC) {
```

Comment:
You can use a `switch` structure to stay on the same page as other methods below.